discovery_engine_core/core/src/engine.rs

```diff
@@ -302,60 +298,66 @@ where
     /// Updates the stacks with data related to the top key phrases of the current data.
     ///
     /// Requires a threshold below which new items will be requested for a stack.
-    async fn update_stacks(
-        &mut self,
-        history: &[HistoricDocument],
-        request_new: usize,
-    ) -> Result<(), Error> {
+    #[instrument(skip(self))]
+    async fn update_stacks(&mut self, history: &[HistoricDocument], request_new: usize) {
         let mut stacks = self.stacks.write().await;
-
-        let mut stacks: Vec<_> = stacks
+        let stacks = stacks
```

Comment:
it's more clear ihmo to have `:Vec<_>` here then having to go to look at the and of a multiline expression. It also reduce the changes. same below

Reply:
i'll revert it for the sake of fewer changed lines.

after all this is highly subjective, i for example find it much more readable if the `collect()` says what type is collected via the turbufish, otherwise i have to go back to the assignment to read the type after i have read the multiline operation.
also the type of the bound variable usually is displayed in the IDE via the rust-analyzer anyways. if we then write `: Vec<_>` we actually see less information, because the IDE doesn't give a type hint anymore, which would have also contained information about the `_` inferred type.

Reply:
mmmm, I didn't know about this difference in the ide.